File: java_project/oops/Banking/Customer.java
package oops.Banking;



public abstract class Customer {
    private String name;
    private long customerId;
    private Account account;

    private static long idCounter = 10001L;

    public Customer (String name){
        this.name = name;
        this.customerId =idCounter++;
        this.account = new Account (this.customerId+900000L);
    }
    public String getName (){
        return name;
    }
    public long getCustomerId(){
        return customerId;
    }
    public Account getAccount(){
        return account;
    }
    
}

File: js_project/oops/Banking/Customer.js
const Account = require('./Account');

class Customer{
    static #idCounter = 10001;

    #name
    #customerId;
    #account;

    constructor(name){
        if (new.target === Customer) {
            throw new Error("Cannot instantiate an abstract class directly.");
        }

        this.#name = name;
        this.#customerId = customerId.#idCounter;
        customerId.idCounter++;

        this.#account = new Account (this.#customerId + 900000);

    }

    getName(){
        return this.#name;
    }

    getCustomerId(){
        return this.#customerId;
    }

    getAccount(){
        return this.#account;
    }
}

module.export = Customer;